### PR TITLE
Check for origin access before allowing updates to an origin

### PR DESCRIPTION
**Pull request by reset**
tenor-66397537

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/handlers/integrations.rs [hunk 1]
```diff
@@ -79,6 +79,10 @@ pub fn fetch_origin_integration_names(req: &mut Request) -> IronResult<Response>
         Ok(p) => p,
         Err(st) => return Ok(Response::with(st)),
     };
+    let session = req.extensions.get::<Authenticated>().unwrap().clone();
```
Comment (chefsalim) [inline components/builder-depot/src/handlers/integrations.rs]: we already check origin access in the validate_params helper

File: components/builder-depot/src/handlers/integrations.rs [hunk 2]
```diff
@@ -106,6 +110,10 @@ pub fn create_origin_integration(req: &mut Request) -> IronResult<Response> {
...
```
Comment (chefsalim) [inline components/builder-depot/src/handlers/integrations.rs]: same as above

File: components/builder-depot/src/handlers/integrations.rs [hunk 3]
```diff
@@ -155,6 +163,10 @@ pub fn delete_origin_integration(req: &mut Request) -> IronResult<Response> {
...
```
Comment (chefsalim) [inline components/builder-depot/src/handlers/integrations.rs]: same as above

File: components/builder-depot/src/handlers/integrations.rs (hunk 1 above)
Comment (reset) [reply components/builder-depot/src/handlers/integrations.rs]: Ah of course! Updated the PR :)

**Pull request merged**